packages/headless/src/features/commerce/facets/facet-set/facet-set-slice.ts

```diff
@@ -122,6 +128,27 @@ export const commerceFacetSetReducer = createReducer(
         updateExistingFacetValueState(existingValue, 'select');
         facetRequest.freezeCurrentValues = true;
       })
+      .addCase(toggleSelectLocationFacetValue, (state, action) => {
+        const {facetId, selection} = action.payload;
+        const facetRequest = state[facetId]?.request;
+
+        if (!facetRequest || !ensureLocationFacetRequest(facetRequest)) {
+          return;
+        }
+
+        facetRequest.preventAutoSelect = true;
+
+        const existingValue = facetRequest.values.find(
+          (req) => req.value === selection.value
+        );
+        if (!existingValue) {
+          insertNewValue(facetRequest, selection);
+          return;
+        }
+
+        updateExistingFacetValueState(existingValue, 'select');
+        facetRequest.freezeCurrentValues = true;
```

Comment:
Same for freezeCurrentValues. It's removed.

Reply:
Will tackle this in https://github.com/coveo/ui-kit/pull/4574

Reply:
The values will also be displayed either in order of distance or alphabetically, so there should be too much jumping.